### ISSUE TITLE
8.9 no longer exports string parsing scope

### ISSUE THREAD

**Issue by HuStmpHrrr** [coq/coq#9707]
#### Description of the problem

8.9 no longer exports String by default. in 8.8.2 or ealier, text between double quotes will become string without having to import `String`, but 8.9 doesn't do it anymore. this breaks many libraries, in which string might be used to, for example, indicate which case the current bullet point is dealing with.

#### Coq Version

8.9

**Comment by HuStmpHrrr** [comment]
I guess one followup question is, how can I just export the parsing scope everywhere without exporting the functions in the library?

**Comment by JasonGross** [comment]
You can do
```coq
Require Coq.Strings.String.
Export String.StringSyntax.
```

I'm inclined to close this bug report as either INVALID or WONTFIX; note that this change is mentioned in [CHANGES](https://github.com/coq/coq/blob/master/CHANGES.md):
> Syntax notations for `string`, `ascii`, `Z`, `positive`, `N`, `R`,
  and `int31` are no longer available merely by `Require`ing the files
  that define the inductives.  You must `Import` `Coq.Strings.String.StringSyntax`
  (after `Require` `Coq.Strings.String`), `Coq.Strings.Ascii.AsciiSyntax` (after
  `Require` `Coq.Strings.Ascii`), `Coq.ZArith.BinIntDef`, `Coq.PArith.BinPosDef`,
  `Coq.NArith.BinNatDef`, `Coq.Reals.Rdefinitions`, and
  `Coq.Numbers.Cyclic.Int31.Int31`, respectively, to be able to use
  these notations.  Note that passing `-compat 8.8` or issuing
  `Require Import Coq.Compat.Coq88` will make these notations
  available.  Users wishing to port their developments automatically
  may download `fix.py` from
  <https://gist.github.com/JasonGross/5d4558edf8f5c2c548a3d96c17820169>
  and run a command like `while true; do make -Okj 2>&1 |
  /path/to/fix.py; done` and get a cup of coffee.  (This command must
  be manually interrupted once the build finishes all the way though.
  Note also that this method is not fail-proof; you may have to adjust
  some scopes if you were relying on string notations not being
  available even when `string_scope` was open.)

**Comment by HuStmpHrrr** [comment]
thx. I will come back to this if I have trouble doing the migration.

**Comment by HuStmpHrrr** [comment]
@JasonGross it gives this error though:

> Error: String.StringSyntax is not a module

I am pretty sure I am working with 8.9:

```
$ coqc -v
The Coq Proof Assistant, version 8.9.0 (February 2019)
compiled on Feb 20 2019 17:36:49 with OCaml 4.05.0
```

what could happen in this case?

**Comment by HuStmpHrrr** [comment]
I guess the explanation is, in 8.9 stdlib, there isn't such module?
https://github.com/coq/coq/blob/v8.9/theories/Strings/String.v

in master, it does though:
https://github.com/coq/coq/blob/master/theories/Strings/String.v#L487

**Comment by HuStmpHrrr** [comment]
Is there any chance there can be a fix at 8.9.1 (or whatever version comes next)?

I had to rollback to 8.8.2 if there is no effective fix for all dependencies.

**Comment by JasonGross** [comment]
You can either pass `-compat 8.8` on the command line, or you can make your own module with
```coq
Module Export StringSyntax. Declare ML Module "string_syntax_plugin". End StringSyntax.
Module Export AsciiSyntax. Declare ML Module "ascii_syntax_plugin". End AsciiSyntax.
```
and then export that everywhere.

In fact, https://github.com/coq/coq/pull/8795 is one of the 44 PRs categorized under "Request Inclusion in 8.9.1" in the [Coq 8.9 Project](https://github.com/coq/coq/projects/29#card-14724234).  I'm not sure when 8.9.1 is slated to be released (@silene ?), but I believe we plan to release the first beta version of Coq 8.10 at the beginning of May.
